packages/metro-memory-fs/src/index.js

```diff
@@ -675,19 +675,22 @@ class MemoryFs {
     }
   };
 
-  mkdirSync: (dirPath: string | Buffer, mode?: number) => void = (
+  mkdirSync: (
     dirPath: string | Buffer,
-    mode?: number,
+    mode?: number | {recursive?: boolean, mode?: number},
```

Comment:
This API change seems unused and unrelated to the dependency bump. What's the thinking here?

Reply:
Great question. This one caught me by surprise when I made the change too.

metro-memory-fs is a shim for the nodejs `fs` package. It is used a lot in the tests at the very least. 

mpdirp used to do the recursive directory creation manually but in node 10, the `require('fs').mkdir` api changed to add the recursive flag. So the mkdirp package uses that now instead. The (very good and thorough) tests were then failing because this library is shim'ing nodejs and thus the code is no longer working.

This change to mkdirp mirrors the API from node 10's mkdir, as far as is needed for this repo's use of mkdirp.

Reply:
Ah, got it, thanks - I didn't realise this was a shim for `fs`. In that case I'd just rename `mode` -> `options` and `realMode` -> `mode`.